source/Octopus.Tentacle.Tests.Integration/ClientScriptExecutionScriptServiceV2IsNotRetried.cs

```diff
@@ -0,0 +1,231 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut;
+using NUnit.Framework;
+using Octopus.Tentacle.CommonTestUtils.Builders;
+using Octopus.Tentacle.Contracts;
+using Octopus.Tentacle.Contracts.ScriptServiceV2;
+using Octopus.Tentacle.Tests.Integration.Support;
+using Octopus.Tentacle.Tests.Integration.Util;
+using Octopus.Tentacle.Tests.Integration.Util.Builders;
+using Octopus.Tentacle.Tests.Integration.Util.Builders.Decorators;
+using Octopus.Tentacle.Tests.Integration.Util.TcpTentacleHelpers;
+
+namespace Octopus.Tentacle.Tests.Integration
+{
+    [IntegrationTestTimeout]
+    public class ClientScriptExecutionScriptServiceV2IsNotRetried : IntegrationTest
+    {
+        [Test]
+        [TestCaseSource(typeof(TentacleTypesToTest))]
+        public async Task WhenANetworkFailureOccurs_DuringStartScript_WithRetriesDisabled_TheCallIsNotRetried(TentacleType tentacleType)
+        {
+            using var clientTentacle = await new ClientAndTentacleBuilder(tentacleType)
+                .WithRetriesDisabled()
+                .WithPortForwarderDataLogging()
+                .WithResponseMessageTcpKiller(out var responseMessageTcpKiller)
+                .WithRetryDuration(TimeSpan.FromMinutes(4))
+                .WithTentacleServiceDecorator(new TentacleServiceDecoratorBuilder()
+                    .LogCallsToScriptServiceV2()
+                    .RecordExceptionThrownInScriptServiceV2(out var scriptServiceExceptions)
+                    .CountCallsToScriptServiceV2(out var scriptServiceCallCounts)
+                    .DecorateScriptServiceV2With(new ScriptServiceV2DecoratorBuilder()
+                        .BeforeStartScript(() =>
+                        {
+                            if (scriptServiceExceptions.StartScriptLatestException == null)
+                            {
+                                responseMessageTcpKiller.KillConnectionOnNextResponse();
+                            }
+                        })
+                        .Build())
+                    .Build())
+                .Build(CancellationToken);
+
+            var waitForFile = Path.Combine(clientTentacle.TemporaryDirectory.DirectoryPath, "waitforme");
+
+            var startScriptCommand = new StartScriptCommandV2Builder()
```

Comment:
You can delete 

```
.WithIsolation(ScriptIsolationLevel.FullIsolation)
.WithMutexName("bob")
```

As this test does not require them and is not trying to assert any behaviour around script isolation or mutex locks

Reply:
This is part of ensuring all scripts are completed , which may not be needed as per your [other comment](https://github.com/OctopusDeploy/OctopusTentacle/pull/534#discussion_r1252405871).

I'm not sure about this in general, so I'll get some more info from @LukeButters